server/src/main/java/org/elasticsearch/ingest/IngestService.java

```diff
@@ -875,17 +907,20 @@ private void executePipelines(
                 } catch (IllegalArgumentException ex) {
                     // An IllegalArgumentException can be thrown when an ingest processor creates a source map that is self-referencing.
                     // In that case, we catch and wrap the exception, so we can include more details
-                    listener.onFailure(
-                        new IllegalArgumentException(
-                            format(
-                                "Failed to generate the source document for ingest pipeline [%s] for document [%s/%s]",
-                                pipelineId,
-                                indexRequest.index(),
-                                indexRequest.id()
-                            ),
-                            ex
-                        )
+                    Exception documentContainsSelfReferenceException = new IllegalArgumentException(
```

Comment:
I was confused by this exception name for a moment, until I read the comment. We could probably use a smaller name for it if we wanted :)